nodejs/webapi/swagger/openapiservice.js

```diff
@@ -0,0 +1,58 @@
+const swaggerJsdoc = require('swagger-jsdoc');
+const swaggerUi = require('swagger-ui-express');
+
+// Basic swagger definition
+const BasicswaggerDefinition = {
+    definition: {
+        openapi: "3.0.0",
+        info: {
+         title: "ApiApp",
+         description: "[http://localhost:3030/swagger.json](http://localhost:3030/swagger.json)",
+         version: "1.0.0"
+        },
+    },
+    apis: ['./server.js'],
+};
+
+// Improved swagger definition
+const ImprovedswaggerDefinition = {
+    definition: {
+        openapi: "3.0.0",
+        info: {
+         title: "Weather Forecast API",
+         description: "[http://localhost:3030/swagger.json](http://localhost:3030/swagger.json)",
+         version: "1.0.0",
+         contact: {
+            name: 'Contoso',
+            email: 'api@contoso.com',
+            url: 'https://contoso.com/support',
+         },
+        },
+        servers: [
+           {
+            url: 'http://localhost:3030',
+           },
+        ],
+        tags: [
+           {
+            name: 'Weather',
+            description: 'Weather API',
+           },
+        ],
+    },
+    apis: ['./server.js'],
+  };
+
+const swaggerSpecs = {
+  basic: swaggerJsdoc(BasicswaggerDefinition),
+  improved: swaggerJsdoc(ImprovedswaggerDefinition),
+};
+
+const setupSwaggerUi = (app, specs, path = '/api-docs') => {
```

Comment:
Would be able to change the Swagger UI endpoint from `/api-docs` to either `/swagger` or `/api-docs/swagger`? Semantically, it's closer to `/swagger`, though.

Also, can it only be exposed if it's running on my local machine? It shouldn't be exposed when it's deployed to the cloud.

Reply:
Done.

This, as well as the configuration for https is done on the infra side (Azure), and not setup on the code side.